lib/sync-settings.coffee

```diff
@@ -8,33 +8,97 @@ _ = require 'underscore-plus'
 DESCRIPTION = 'Atom configuration storage operated by http://atom.io/packages/sync-settings'
 REMOVE_KEYS = ["sync-settings"]
 
-module.exports =
+SyncSettings =
   config: require('./config.coffee')
 
   activate: ->
-    GitHubApi ?= require 'github'
-    PackageManager ?= require './package-manager'
-    Tracker ?= require './tracker'
-
-    atom.commands.add 'atom-workspace', "sync-settings:backup", =>
-      @backup()
-      @tracker.track 'Backup'
-    atom.commands.add 'atom-workspace', "sync-settings:restore", =>
-      @restore()
-      @tracker.track 'Restore'
-    atom.commands.add 'atom-workspace', "sync-settings:view-backup", =>
-      @viewBackup()
-      @tracker.track 'View backup'
-
-    # make the tracking last in case any exception happens
-    @tracker = new Tracker 'sync-settings._analyticsUserId', 'sync-settings.analytics'
-    @tracker.trackActivate()
+    # speedup activation by async initializing
+    setImmediate =>
+      # actual initialization after atom has loaded
+      GitHubApi ?= require 'github'
+      PackageManager ?= require './package-manager'
+      Tracker ?= require './tracker'
+
+      atom.commands.add 'atom-workspace', "sync-settings:backup", =>
+        @backup()
+        @tracker.track 'Backup'
+      atom.commands.add 'atom-workspace', "sync-settings:restore", =>
+        @restore()
+        @tracker.track 'Restore'
+      atom.commands.add 'atom-workspace', "sync-settings:view-backup", =>
+        @viewBackup()
+        @tracker.track 'View backup'
+
+      @checkForUpdate() if atom.config.get('sync-settings.checkForUpdatedBackup')
+
+      # make the tracking last in case any exception happens
```

Comment:
What kind of exceptions are you expecting here? I think they should be handled explicitly since it looks like if the `@tracker` variable is undefined (due to such an exception?) the other code doesn't check for it which might be responsible for #155.